### ISSUE TITLE
Holopy calculates holograms incorrectly for some/many scattering geometries

### ISSUE THREAD

**Issue by briandleahy** [manoharan-lab/holopy#283]
The current ```calc_holo``` pipeline in ```holopy.scattering.calculations``` involves calling ```scattered_field_to_hologram```, which basically consists of this line:
```python
    holo = (np.abs(scat+ref)**2 * (1 - normals)).sum(dim=vector)
    return holo
```
This is clearly wrong -- ```1 - normals``` is not coordinate invariant. I believe what this line is trying to do is evaluate the Poynting vector, by assuming that the Poynting vector is always  E_x^2 + E_y^2, which (i) it is not (see below) and (ii) this line does not do. Two quick obvious examples why the line fails to evaluate E_x^2 + E_y^2, even when ```normals``` is normalized (which is not checked):
1. Consider the case where ```normals = (0, 0, -1)```. Then ```1 - normals = (1, 1, 2)```, and the function returns Ex^2 + Ey^2 + 2Ez^2.
2. Consider the case where ```normals = (-1, 0, 0)```. Then ```1 - normals = (2, 1, 1)```, and the function returns 2Ex^2 + Ey^2 + Ez^2. For a typical hologram, (Ex,Ey, Ez) ~ (Ex, 0, 0), so this gives a hologram which is roughly 2x what it should be.

(Incidentally, I found these behaviors by trying to generate a hologram through ```detector_points```, which currently defaults to a normal pointing in the -z direction and which gives normals approaching (-1, 0, 0) at large rho, and wondering why my hologram went to 2 at large values.)

Fortunately, this expression happens to be mostly correct (but ***still wrong***, see below) for the usual use case of a detector that is oriented along the *+z* direction. When ```normals = (0, 0, 1)```, then ```1 - normals = (1, 1, 0)```, and this returns Ex^2 + Ey^2.

One option would be to fix this buggy line by just replacing this with something like
```python
    total_field = scat + ref
    # assume the field propagates in -z
    poynting_magnitude = np.abs(total_field[:, 0])**2 + np.abs(total_field[:, 1])**2
    poynting_direction = np.array([0, 0, -1])
    poynting_flux = poynting_direction.dot(normals) * poynting_magnitude
    return poynting_flux
```

But this is **still wrong**. Why? The measured hologram on the detector should be the Poynting flux into to the detector, which depends on *both* the electric and magnetic fields, or equivalently the electric field and the wavevector. The original calculation pipeline was assuming that the wavevector is always along the -z direction, so then the Poynting vector is just Ex^2 + Ey^2 (although the code is in general wrong). However, the Poynting vector is not always along the -z direction, even if the Poynting vector of the incident light is along the -z direction. For a scattering geometry, the total field is the scattered plus the incident field, so the Poynting vector will contain terms from the scattered field which will shift the Poynting vector away from (0, 0, -1) if the detection angle is not directly below the particle. 

Fortunately, though, this correction is in general small -- for pixels directly below the particle, the Poynting flux is proportional to Ex^2 + Ey^2, and for pixels far away from the particle's in-plane position, the scattering angle is necessarily large, and S(\theta) is in general small. 

Still, the current expression is buggy and incorrect and should be fixed. 

I see 3 options:
1. Fix the current code to always use the Poynting vector as (0, 0, Ex^2 + Ey^2), regardless of the position on the detector (*e.g.* something like I have above). This might seem a little restrictive, but ```holopy``` already assumes that the incident field is propagating along the -z direction, so it's not actually that restrictive. The main problem I have with it however is that it's still wrong, as it ignores the change in the Poynting vector's orientation with position on the detector.
2. Have ```scattered_field_to_hologram``` calculate the correct Poynting vector. This could be done by assuming that the fields scattered from the particle are roughly in the form e^{ikr} / r and using that to calculate a Poynting vector. The main disadvantage I see with this is that a pipeline that works for a hologram in the lens-free appproximation would **not** work for a hologram modeled with a lens. Thus, this option will make @briandleahy and @ralex0 very sad.
3. Have the ```ScatteringTheory``` know how to calculate a hologram and/or a Poynting vector. I think this is the long-term correct solution, but it would be a bit of extra work, as ScatteringTheory would need to get partially overhauled.

As far as I can tell, this line has been like this for quite a long time (at least 3 years). So I'm assuming there was some sort of design choice with this done previously and/or wondering if there is some physics which I'm missing. Perhaps the older ```holopy``` generation, such as @tdimiduk  and @jeromefung  can weigh in? Am I missing something completely here?

And @vnmanoharan @barkls @briandleahy @ralex0  -- I've assigned this issue to all of us just because I think whatever we do should come from consensus with the 4 of us. So, what are your thoughts?

**Comment by briandleahy** [comment]
I'll go first. My preferences are Option (1) as a stopgap, then option (3) for longer-term development.

**Comment by vnmanoharan** [comment]
I don't think this was a design decision. There was always a tension between supporting analysis of in-line holograms and supporting more general geometries. I think all of our testing went into the in-line geometry, and other geometries haven't been tested.  With the in-line geometry, there are a few assumptions:
1) the transmitted beam is the reference
2) the detector is centered on the transmitted beam and is oriented normal to it
3) the detector is sensitive only to the z-component of the Poynting vector
4) the detector extends to only small angles about the forward direction

The third assumption is why we calculate Ex^2 + Ey^2 at all pixels on the detector, even away from forward scattering.

[Jerome's thesis](https://dash.harvard.edu/handle/1/11181215) (pp. 36-37) discusses the effects of relaxing the fourth assumption.

**Comment by barkls** [comment]
I think the current calculation is fine for calculating regular holograms, since any deviation of the Poynting vector from the z-axis will be very small for typical imaging setups (focal plane >= 10 um from particle, hologram extends +/- 5um in x and y). However, the current implementation with the `1 - normals` is messy and should be fixed.

The current calculation doesn't work at all for hologram calculations with `detector_points`. The default normals of `detector_points` used to also be in the z-direction, but I changed them to point radially inwards in 9f9322e849d7587d7cdaebc94f0a8598fc38adff and  69b0c7bd56175a8e1abb58eea32995f7935ec15c. However, I didn't update the actual hologram calculation so now it just spits out nonsense. I still think it makes most sense for default normals of `detector_points` to be radially inwards. We could implement Brian's solution 2 for this (it wouldn't disrupt MieLens since that doesn't work for `detector_points` anyway), or we could consider dropping support of using `detector_points` for a hologram calculation and just have it available for other scattering calculations.

**Comment by briandleahy** [comment]
I calculated what the difference is in a hologram as calculated the correct way, with the Poynting vector, and the current way in holopy, by taking |E_x|^2 + |E_y|^2. While there is a difference, as I said before, for typical geometries the difference is minuscule -- I estimate that for a typical hologram the difference is about half a percent. So as @barkls and @vnm were saying, this difference is practically irrelevant.

However, the calculation for normals other than ```[0, 0, 1]``` is still incorrect. So, pursuant to discussions with @ralex0 and @barkls , what I will do is:
1. LaTeX up the calculation I did, and upload it here and to the group wiki so others can see.
2. Fix the hologram calculation to always return |E_x|^2 + |E_y|^2.
3. Deprecate normals throughout the rest of ```holopy```, since they are only used in ```calc_holo``` and there they are used incorrectly.

I'll wait to do 2 and 3 until 1 is done, so y'all will have plenty of time to let me know if this is a bad direction to go.

**Comment by briandleahy** [comment]
Here's the TeX'd up derivation. I can put it on the group wiki if desired as well.

I will go ahead and fix the hologram calculation to (2) always return |E_x^2 + |E_y|^2 and (3) deprecrate normals throughout the rest of holopy.
[poynting_vector_vs_intensity.pdf](https://github.com/manoharan-lab/holopy/files/3526126/poynting_vector_vs_intensity.pdf)
